Fix build and tests using the most recent stable version of Rust compiler and libs/deps

ekutt opened the pull request.

ekutt committed:
Fix build and tests using the most recent stable version of Rust compiler and libs/deps

ekutt commented on Cargo.toml:
```diff
@@ -17,19 +17,18 @@ mio-extras = "~2.0.5"
-#git = "https://github.com/maidsafe/socket-collection"
-#branch = "community-release-1"
+version = "~0.4.0"
+git = "https://github.com/ekutt/socket-collection"
```
We also need to update some dependencies in maidsafe/socket-collection repo which is already archived. Maybe we need to fork it and merge the changes into our repo.

knocte replied:
forked into /nblockchain/ org thanks, you can update the PR

ekutt replied:
> forked into /nblockchain/ org thanks, you can update the PR

Thank you! Created a pull request for that also.

knocte replied:
What pull request? I thought this had to just change in this PR, I don't understand

ekutt replied:
There are two dependencies (maidsafe_utilities, safe_crypto) that need to be updated in socket-collection repo also, otherwise it doesn't build. After it gets merged into nblockchain/socket-collection, I will update the socket-collection dependency here to reference to nblockchain/socket-collection repo:
https://github.com/nblockchain/socket-collection/pull/1

knocte commented:
can you add a quick githubCI lane to run the build and tests please

ekutt committed:
Update .travis.yml

ekutt committed:
Update .travis.yml

ekutt committed:
Update .travis.yml

knocte commented:
I said GithubCI, not travis :( Travis is not cross-platform.

ekutt committed:
Create rust.yml

ekutt committed:
Create rust.yml

ekutt commented:
> I said GithubCI, not travis :( Travis is not cross-platform.

Oh, sorry, my mistake.

ekutt committed:
Update README.md

ekutt commented:
I have added a GithubCI workflow: 
https://github.com/ekutt/p2p/runs/484681414?check_suite_focus=true

knocte commented on .github/workflows/rust.yml:
```diff
@@ -0,0 +1,15 @@
+name: Rust
+
+on: [push]
```
can you change this to `[push, pull_request]` please? this way the CI status will show up in this PR too

ekutt replied:
> can you change this to `[push, pull_request]` please? this way the CI status will show up in this PR too

Done.

knocte commented on .github/workflows/rust.yml:
```diff
@@ -0,0 +1,15 @@
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: cargo build --verbose
```
mmm seems you didn't need to setup Rust in the CI lane; then it must mean that GithubActions VMs already include it; but how can we make sure the version used is the stable one?

ekutt replied:
I am not sure, I will check how to make sure the version used is the stable one.

knocte replied:
let's print version number in the console at least

ekutt replied:
> let's print version number in the console at least

Done. Seems that it is Rust 1.41.1 and Cargo 1.41.0.

ekutt replied:
I modified rust.yml to setup the rust to stable version.

ekutt committed:
Update rust.yml

ekutt committed:
Update Cargo.toml

Update socket-collection dependency to reference to nblockchain/socket-collection repo.

ekutt committed:
Update README.md

knocte commented on README.md:
```diff
@@ -2,7 +2,7 @@
 |Crate|Documentation|Linux/OSX/Windows|
 |:---:|:-----------:|:----------------:|
-| [![](http://meritbadge.herokuapp.com/p2p)](https://crates.io/crates/p2p) | [![Documentation](https://docs.rs/p2p/badge.svg)](https://docs.rs/p2p) | [![Build Status](https://travis-ci.org/ustulation/p2p.svg?branch=master)](https://travis-ci.org/ustulation/p2p)
+| [![](http://meritbadge.herokuapp.com/p2p)](https://crates.io/crates/p2p) | [![Documentation](https://docs.rs/p2p/badge.svg)](https://docs.rs/p2p) | [![Rust](https://github.com/ekutt/p2p/workflows/Rust/badge.svg?branch=P2P-1_FIX_BUILD_AND_TESTS)](https://github.com/ekutt/p2p/workflows/Rust/badge.svg?branch=P2P-1_FIX_BUILD_AND_TESTS)
```
s/ekutt/nblockchain

ekutt replied:
Done.

ekutt committed:
Update README.md

ekutt committed:
Update rust.yml

Add step to print Rust version

ekutt committed:
Update rust.yml

Add script to set up Rust to stable version.

ekutt committed:
Update README.md

ekutt committed:
Merge branch 'master' into P2P-1_FIX_BUILD_AND_TESTS
